File: GPS/app/src/main/java/study/gps/MainActivity.java
package study.gps;

import android.Manifest;
import android.app.Fragment;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Build;
import android.support.annotation.NonNull;
import android.support.v4.app.ActivityCompat;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import java.security.Permission;

public class MainActivity extends AppCompatActivity {
    public static final int MY_ACCESS_FINE_LOCATION_KEY = 100;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if(ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION)== PackageManager.PERMISSION_DENIED){
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                requestPermissions(new String[]{Manifest.permission.ACCESS_FINE_LOCATION},MY_ACCESS_FINE_LOCATION_KEY);
            }
        }else{
            Intent intent = new Intent(this,MapsActivity.class);
            startActivity(intent);
        }

    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        switch (requestCode){
            case MY_ACCESS_FINE_LOCATION_KEY:
                if(grantResults.length>0&&grantResults[0]==PackageManager.PERMISSION_GRANTED){
                    Log.e("asdf","asdf");
                    Intent intent = new Intent(this,MapsActivity.class);
                    startActivity(intent);
                }
        }
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }
}

File: PracticeBroadCast/app/src/main/java/practice/practicebroadcast/MainActivity.kt
package practice.practicebroadcast

import android.content.Intent
import android.content.IntentFilter
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    init {
        var intent: Intent
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        intent = Intent(this@MainActivity, MyService::class.java)
    }

    fun start(view: View) {
        startService(intent)
    }

    fun stop(view: View) {
        stopService(intent)
    }
}
